rhizome/host/lib/vm_setup.rb

```diff
@@ -421,18 +422,28 @@ def cloudinit(unix_user, public_keys, gua, nics, swap_size_bytes, boot_image)
     end.join("\n")
 
     raparams = nics.map { "ra-param=#{_1.tap}" }.join("\n")
-
+    interfaces = nics.map { "interface=#{_1.tap}" }.join("\n")
+    dnsmasq_address_ip6 = NetAddr::IPv6.parse("fd00:0b1c:100d:53::")
     vp.write_dnsmasq_conf(<<DNSMASQ_CONF)
 pid-file=
 leasefile-ro
 enable-ra
 dhcp-authoritative
+domain-needed
+bogus-priv
+no-resolv
 #{raparams}
+#{interfaces}
 dhcp-range=#{guest_network.nth(2)},#{guest_network.nth(2)},#{guest_network.netmask.prefix_len}
 #{private_ip_dhcp}
-dhcp-option=option6:dns-server,2620:fe::fe,2620:fe::9
-dhcp-option=option:dns-server,149.112.112.112,9.9.9.9
+server=149.112.112.112
+server=9.9.9.9
+server=2620:fe::fe
+server=2620:fe::9
+dhcp-option=option6:dns-server,#{dnsmasq_address_ip6}
+listen-address=#{dnsmasq_address_ip6}
 dhcp-option=26,1400
+bind-interfaces
```

Comment:
I do find the number of options being shuffled around here a bit confusing, it seems like you are making dnsmasq more selective from the commit message though, right?  What's dnsmasq's design theory here?

Reply:
Yes, so, the first 3 options are to handle dns lookups in a safe way, not forwarding dns lookups without dots, not leaking private ip addresses and not using resolve.conf for dns addresses. (domain-needed, bogus-priv, no-resolv)

`interface=`, `listen-address` and `bind-interfaces` part is where it gets interesting. We tell dnsmasq to listen in which interface by `interface` and which address by `listen-address`. However, it still binds itself to all of the interfaces and available addresses but filters the queries. Specifying `bind-interfaces` enforces dnsmasq to bind to the specified interface and address, eliminating the need to filter the queries.